src/main/java/ca/corefacility/bioinformatics/irida/pipeline/results/impl/SISTRSampleUpdater.java

```diff
@@ -93,8 +93,9 @@ public void update(Collection<Sample> samples, AnalysisSubmission analysis) thro
 
 				//loop through each of the requested fields and save the entries
 				SISTR_FIELDS.entrySet().forEach(e -> {
-					if (result.containsKey(e.getKey()) && result.get(e.getKey()) != null) {
-						String value = result.get(e.getKey()).toString();
+					if (result.containsKey(e.getKey())) {
+						Object valueObject = result.get(e.getKey());
+						String value = (valueObject != null ? valueObject.toString() : "");
```

Comment:
I'm a bit confused about what's going on here with the ternary operator.  Is it just stopping `null` from being written to a value?  What is a case that there would be an entry in the `result` map but the value would be `null`?

Reply:
Currently, `null` would never be written to a sample, it would just skip over the entry.  However, this becomes problematic when updating the **cgMLST Type**, since this can be `null`. Skipping over `null` does not make sense here since it will then just leave the old **cgMLST Type** in the linelist, which may now mismatch the updated **Serovar**.

This line, specifically, is blanking out the field (cgMLST Type) in case of `null` instead of writing `null` (this is so it matches what you normally get as output from SISTR, which is an empty field instead of `null` displayed).

Reply:
Got it.  Thanks.